File: backend/routers/transactions/transaction.py
from pydantic import BaseModel

class Transaction(BaseModel):

    id : int = None
    amount : float
    vendor : str = None
    category : str


    def set_id(self, id):
        self.id = id

    def __str__(self):
        return f"""id: {self._id}
amount: {self._amount}
vendor: {self._vendor}
category: {self._category}"""

File: frontend/src/components/Transaction/Transaction.js
import './Transaction.css';

export default function Transaction(props) {
    const transaction = props.transaction

    return (
        <div className={"transaction-container " + (transaction.amount < 0 ? 'negative' : 'non-negative')}>
            <table className='transaction'>
              <tr className='plus'>
                <td>Category: </td>
                <td className='dynam-info text'>{transaction.category}</td>
              </tr>
              <tr>
                <td>Vendor: </td>
                <td className='dynam-info text'>{transaction.vendor}</td>
              </tr>
              <tr>
                <td>Anount: </td>
                <td className='dynam-info number'>{transaction.amount}</td>
              </tr>
              <tr>
                <td colspan="2"><button className='transaction-button' onClick={()=>props.deleteTransaction(transaction.id)}>Delete</button></td>
              </tr>
            </table>
        </div>
    )
}